src/compiler/checker.ts

```diff
@@ -2586,7 +2586,7 @@ namespace ts {
                     let symbolFromVariable: Symbol | undefined;
                     // First check if module was specified with "export=". If so, get the member from the resolved type
                     if (moduleSymbol && moduleSymbol.exports && moduleSymbol.exports.get(InternalSymbolName.ExportEquals)) {
-                        symbolFromVariable = getPropertyOfType(getTypeOfSymbol(targetSymbol), name.escapedText);
+                        symbolFromVariable = getPropertyOfType(getTypeOfSymbol(targetSymbol), name.escapedText, /*skipObjectFunctionPropertyAugment*/ true);

```

Comment:
Actually, rather than threading through this flag, and worrying about the safety of caching it and have two differently calculated symbols for a property lookup, couldn't we

1. Initialize `globalFunctionType` and `globalObjectType` to the empty type (which should naturally cause property lookups that ultimately may have depended on them to fail during global initialization).
2. Check if `symbolFromVariable.parent` here is `globalObjectType` or `globalFunctionType` and, if so, discard the result? (to ensure consistency with the initialization phase)

Reply:
1. We want this to happen irrespective of augmenting that means any time we import the properties from object/function should be error so this might not happen at augmentation time.
2. Will there be some kind of edge case where in valid scenario return globalfunctionType property symbol.